### ISSUE TITLE
Paapi Module: Autoconfig behavior

### ISSUE THREAD

**Issue by lksharma** [prebid/Prebid.js#11242]
## Description
Prebid version  [8.37.0](https://github.com/prebid/Prebid.js/releases/tag/8.37.0) introduced new features allowing auction configurations to be independent of GPT. This version also introduced functionality to clear cached auction configurations by resetting them with null values using setConfig. However, in certain scenarios, cached auction configurations can persist, leading to duplicate impression calls, as observed in version [8.36.0](https://github.com/prebid/Prebid.js/releases/tag/8.36.0). This ticket also provides additional context to [issue #11102](https://github.com/prebid/Prebid.js/issues/11102) which is pending discussions on the desired default autoconfig behaviour.

## Issue Details
- In the latest version of Prebid, with `fledgeForGpt.autoconfig` set to true (default), cached auction configurations are reset correctly.
- However, when `fledgeForGpt.autoconfig` is set to false, additional API calls are required to trigger on-device auctions.
- Specifically, the publisher needs to call `pbjs.setPAAPIConfigForGPT()` whenever pbjs.requestBids is invoked and ensure it's called within the `bidsBackHandler` function.

## Problem Scenario:
If the publisher neglects to call `pbjs.setPAAPIConfigForGPT(`) for subsequent refresh calls, Prebid does not reset the previously set auction configuration, leading to potential duplicate impression calls.

## Example Setup:
```JavaScript

function initAdserver() {
    pbjs.initAdserverSet = true;
    googletag.cmd.push(function() {
        pbjs.que.push(function() {
            pbjs.setPAAPIConfigForGPT();
			...
        });
    });
}

function refreshBid() {
    pbjs.que.push(function() {
        pbjs.requestBids({
            timeout: 1000,
            bidsBackHandler: function() {
                // pbjs.setPAAPIConfigForGPT(); not called
                ...
            }
        });
    });
}
```

## Steps to Reproduce:
- Set fledgeForGpt.autoconfig to false.
- Initiate bids with `pbjs.requestBids` and call `pbjs.setPAAPIConfigForGPT()` in the `bidsBackHandler` function
- For subsequent refresh calls, don't call `pbjs.setPAAPIConfigForGPT()` 
- Observe behaviour

## Potential Solution:
Previously set auction configs are reset by the logic within the [slotConfigurator](https://gitlab.indexexchange.com/parallel-bidding/prebid.js/-/blob/master/modules/fledgeForGpt.js#L29-56) function, which executes when `pbjs.setPAAPIConfigForGPT()` is called. Automating `pbjs.setPAAPIConfigForGPT()` calls within `setTargetingForGPTAsync()` or similar functions could reduce the likelihood of configuration errors and mitigate potential duplicate impression calls.

**Comment by patmmccann** [comment]
> If the publisher neglects to call pbjs.setPAAPIConfigForGPT() for subsequent refresh calls, Prebid does not reset the previously set auction configuration, leading to potential duplicate impression calls.

Another solution is we just tell pubs not to do this bad idea

**Comment by patmmccann** [comment]
@lksharma any objection to closing this and tracking on #11102 ? Or the inverse, closing that and tracking here?

**Comment by lksharma** [comment]
Let's close issue https://github.com/prebid/Prebid.js/issues/11102 and continue discussions within this ticket since it contains most of the relevant context.

**Comment by patmmccann** [comment]
Sounds good, ty!

**Comment by patmmccann** [comment]
One thought: if we start handing auction configs to gam or another TLS before targeting is ready, marrying them together might present some challenges as we try and solve for latency

**Comment by patmmccann** [comment]
Spoke with the GAM team, a 'allAuctionConfigAreSubmitted' function might be coming to GPT. That would make marrying these more difficult if publishers plan to take advantage.

**Comment by patmmccann** [comment]
Consensus - provide this feature request with a new name and delete the autoconfig feature. 
The new name could be ‘submitAuctionConfigsWithTargeting’ or something more compact

**Comment by patmmccann** [comment]
That's covered in #10930

**Comment by patmmccann** [comment]
@lksharma we're chatting about this and realizing we may have achieved your goals; where do we go from here?

**Comment by lksharma** [comment]
Agreed. I've also recently updated the documentation to reflect this case as well: https://github.com/prebid/prebid.github.io/pull/5526. I think we are good to close this out. Thanks.
